### PR TITLE
unparam lint check

### PR DESCRIPTION
**Pull request by mahmud2011**
#### Summary
* add unparam lint check
* fix "\`identifier\` is unused" lint error
* fix "... is always nil" lint err
* fix "always receives ..." lint err
* fix "is never used" lint err

#### Ticket Link
Fixes https://github.com/mattermost/mattermost-server/issues/16622
#### Release Note
```release-note
None
```

### CODE REVIEW

File: app/channel.go [hunk 1]
```diff
@@ -478,7 +478,8 @@ func (a *App) CreateGroupChannel(userIds []string, creatorId string) (*model.Cha
 	return channel, nil
 }
 
-func (a *App) createGroupChannel(userIds []string, creatorId string) (*model.Channel, *model.AppError) {
+// `creatorId` is replaced with _ to pass unused param lint
+func (a *App) createGroupChannel(userIds []string, _ string) (*model.Channel, *model.AppError) {
```
Comment (agnivade) [inline app/channel.go:482]: These are all private methods. I don't believe it needs to satisfy any interface. Why can't we just remove the parameter?

Comment (mahmud2011) [reply app/channel.go:482]: I'll do it next. It's my first task in WIP list.

Comment (agnivade) [reply app/channel.go:482]: Woops, my bad there. Keep going!

**Comment by agnivade** [comment]
Amazing work @mahmud2011 ! Can I ask you to split this PR into small chunks and send them one by one, so that it's easy to review them? We can enable the linter in the last PR.

**Comment by mahmud2011** [comment]
I'll cherry-pick each commit and create separate prs.

**Pull request closed without merging**